### PR TITLE
Update nvidia-driver-installer.yaml to get latest updates for GPU driver installer

### PR DESCRIPTION
**Pull request by elfinhe**
#### What type of PR is this?
To fix test failures on GPU installation in https://github.com/kubernetes/kubernetes/issues/124950.

/kind bug
/kind failing-test

#### What this PR does / why we need it:
Fix GPU test failures in 1.30 release.

#### Which issue(s) this PR fixes:

Fixes # https://github.com/kubernetes/kubernetes/issues/124950

#### Does this PR introduce a user-facing change?

```release-note

```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: test/e2e/testing-manifests/scheduling/nvidia-driver-installer.yaml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 # This DaemonSet was originally referenced from
-# https://github.com/GoogleCloudPlatform/container-engine-accelerators/blob/master/daemonset.yaml
+# https://github.com/GoogleCloudPlatform/container-engine-accelerators/edit/master/nvidia-driver-installer/cos/daemonset-preloaded.yaml
```
Comment (BenTheElder) [inline test/e2e/testing-manifests/scheduling/nvidia-driver-installer.yaml:2]:
```suggestion
# https://github.com/GoogleCloudPlatform/container-engine-accelerators/blob/master/nvidia-driver-installer/cos/daemonset-preloaded.yaml
```

File: test/e2e/testing-manifests/scheduling/nvidia-driver-installer.yaml [hunk 2]
```diff
@@ -80,7 +89,37 @@ spec:
...
+        volumeMounts:
+        - name: nvidia-install-dir-host
+          mountPath: /usr/local/nvidia
+        - name: dev
+          mountPath: /dev
+        - name: nvidia-config
+          mountPath: /etc/nvidia
       containers:
-      - image: "registry.k8s.io/pause:3.9"
+      - image: "gcr.io/google-containers/pause:2.0"
```
Comment (BenTheElder) [inline test/e2e/testing-manifests/scheduling/nvidia-driver-installer.yaml:124]: we should revert this line back, also the upstream source should switch to registry.k8s.io, `grc.io/google-containers` is a) really old b) not actively supported c) GCR is going away next year.

File: test/e2e/testing-manifests/scheduling/nvidia-driver-installer.yaml [hunk 3]
```diff
@@ -47,17 +50,19 @@ spec:
...
+        hostPath:
+          path: /etc/nvidia
       initContainers:
-        # The COS GPU installer image version may be dependent on the version of COS being used.
-        # Refer to details about the installer in https://cos.googlesource.com/cos/tools/+/refs/heads/master/src/cmd/cos_gpu_installer/
-        # and the COS release notes (https://cloud.google.com/container-optimized-os/docs/release-notes) to determine version COS GPU installer for a given version of COS.
-
-        # Maps to gcr.io/cos-cloud/cos-gpu-installer:v2.1.10 - suitable for COS M109 as per https://cloud.google.com/container-optimized-os/docs/release-notes
-      - image: gcr.io/cos-cloud/cos-gpu-installer:v2.1.10
+      - image: "cos-nvidia-installer:fixed"
```
Comment (BenTheElder) [inline test/e2e/testing-manifests/scheduling/nvidia-driver-installer.yaml:60]: is this image available in the cos images we're using?

Comment (elfinhe) [reply test/e2e/testing-manifests/scheduling/nvidia-driver-installer.yaml:60]: Yes, this container image is preloaded on the COS Node image.

File: test/e2e/testing-manifests/scheduling/nvidia-driver-installer.yaml (hunk 2 above)
Comment (elfinhe) [reply test/e2e/testing-manifests/scheduling/nvidia-driver-installer.yaml:124]: Yes, "registry.k8s.io/pause:3.9" is better.

**Comment by BenTheElder** [comment]
If we try this against the release-1.30 branch then we can run some test against the PR, in this branch we don't have real GPU tests currently.

File: test/e2e/testing-manifests/scheduling/nvidia-driver-installer.yaml (hunk 2 above)
Comment (Dragoncell) [reply test/e2e/testing-manifests/scheduling/nvidia-driver-installer.yaml]: +1 on revert back

File: test/e2e/testing-manifests/scheduling/nvidia-driver-installer.yaml (hunk 3 above)
Comment (Dragoncell) [reply test/e2e/testing-manifests/scheduling/nvidia-driver-installer.yaml:60]: From the log. 
`Using image: cos-109-17800-147-22 from project: cos-cloud as node image`. it is actually using the base COS image without preloading. so we should update the exact version for 1.30 used here.

Comment (elfinhe) [reply test/e2e/testing-manifests/scheduling/nvidia-driver-installer.yaml:60]: @Dragoncell I see.

Comment (Dragoncell) [reply test/e2e/testing-manifests/scheduling/nvidia-driver-installer.yaml:60]: From the release notes, v2.1.10 is the right one actually for image `cos-109-17800-147-22`

https://cloud.google.com/container-optimized-os/docs/release-notes/m109#cos-109-17800-147-9_

File: test/e2e/testing-manifests/scheduling/nvidia-driver-installer.yaml [hunk 4]
```diff
@@ -80,7 +89,37 @@ spec:
...
+        - | 
+          echo "Checking for existing GPU driver modules"
+          if lsmod | grep nvidia; then
+            echo "GPU driver is already installed, the installed version may or may not be the driver version being tried to install, skipping installation"
+            exit 0
+          else
+            echo "No GPU driver module detected, installing now"
+            /cos-gpu-installer install
+          fi
+      - image: "gcr.io/gke-release/nvidia-partition-gpu@sha256:e226275da6c45816959fe43cde907ee9a85c6a2aa8a429418a4cadef8ecdb86a"
```
Comment (Dragoncell) [inline test/e2e/testing-manifests/scheduling/nvidia-driver-installer.yaml:106]: without init container of partition-gpus should be fine, as it is not testing the MIG features

**Pull request closed without merging**